### PR TITLE
Update NodeLink component to use FLAGS.CAN_LIST_NODE

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Works as follows:
- if no name, will show `-` as before
- if name, but user cannot list nodes, will show `{name}`
- if renders before FLAG.CAN_LIST_NODE is set, it will initially show `{name}` but will update to a link once the flag is resolved.

### CODE REVIEW

**Comment by benjaminapetersen** [comment]
Updates the pods pages, as well as the pods lists on other pages.

File: frontend/public/components/utils/resource-link.jsx [hunk 1]
```diff
@@ -63,3 +64,16 @@ export const ResourceLink = connectToModel(
...
+const NodeLink_ = (props) => {
+  const {name, flags: { CAN_LIST_NODE: listNodeFlag }} = props;
+  if (!name) {
+    return <span>-</span>;
+  }
+  return listNodeFlag
+    ? <Link to={`/k8s/cluster/nodes/${name}`}>{name}</Link>
+    : <span>{name}</span>;
+};
+
```
Comment (benjaminapetersen) [inline frontend/public/components/utils/resource-link.jsx]: @spadgett already fixed for you, as tempted as I was to do:

```JavaScript 
  return name 
    ? listNodeFlag  
      ? <Link to={`/k8s/cluster/nodes/${name}`}>{name}</Link> 
      : <span>{name}</span>
    : <span>-</span>;
```
😄

File: frontend/public/components/utils/resource-link.jsx [hunk 2]
```diff
@@ -63,3 +64,16 @@ export const ResourceLink = connectToModel(
...
+const NodeLink_ = (props) => {
+  const {name, flags: { CAN_LIST_NODE: listNodeFlag }} = props;
+  if (!name) {
+    return <span>-</span>;
+  }
+  return listNodeFlag
+    ? <Link to={`/k8s/cluster/nodes/${name}`}>{name}</Link>
+    : <span>{name}</span>;
+};
```
[the remark this answers is not in the document]
Comment (benjaminapetersen) [reply frontend/public/components/utils/resource-link.jsx]: Yeah I will add the `<React.Fragment>`, thanks!

**Comment by benjaminapetersen** [comment]
Sure can, am fishing around for other instances now.

**Comment by benjaminapetersen** [comment]
@tlwu2013 we talked about this yesterday.  Would you have issues with changing the `namespace` icon:

screen shot 2018-07-20 at 10 36 53 am

to `NS` so we can use `N` for `node`?  Currently we don't put an icon next to `node` at all.

**Comment by benjaminapetersen** [comment]
@tlwu2013 so we do have an icon for `node`, but its a bit heavy:

screen shot 2018-07-20 at 10 41 13 am

**Comment by benjaminapetersen** [comment]
So proposing `NS` for `namespace`, and `N` for `node`:

screen shot 2018-07-20 at 10 50 20 am

File: frontend/public/components/utils/resource-link.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (benjaminapetersen) [reply frontend/public/components/utils/resource-link.jsx]: Vs the `<span>`?  Hmm. I've tended towards `<React.Fragment>` as a bit of a hack fix for when you want to return `> 1` siblings without a wrapper.  Seems like a `<span>` is fine here, no?

[the remark this answers is not in the document]
Comment (benjaminapetersen) [reply frontend/public/components/utils/resource-link.jsx]: Unless the raw text node is preferrable.

[the remark this answers is not in the document]
Comment (benjaminapetersen) [reply frontend/public/components/utils/resource-link.jsx]: Looking at adding in the `ResourceIcon`, honestly I think the only reason that `<ResourceLink>` was not used was to avoid the BIG `NOD` icon... otherwise there doesn't seem to be a point in having a separate `<NodeLink>` component.

So, if @tlwu2013 is cool with the swap `N` and `NS`, This component can be deleted.

[the remark this answers is not in the document]
Comment (spadgett) [reply frontend/public/components/utils/resource-link.jsx]: Using the span creates unnecessary span elements in the markup that's generated, whereas React.Fragement doesn't. So I'd prefer the fragment to the span.

There's syntactic sugar where you can do `<>foo</>` without needing to type React.Fragment. I think it might cause problems for eslint, though. We should look at that.

Here I'd prefer the raw text node, though.

**Comment by spadgett** [comment]
@tlwu2013 We were guessing that we don't show the node resource icon because it's so similar to namespace icon and they tend to be near each other. Is that the reason?

File: frontend/public/components/utils/resource-link.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (spadgett) [reply frontend/public/components/utils/resource-link.jsx]: > ... This component can be deleted.

I don't think we can... You still need to connectToFlags to check if you can list the nodes. I'd rather not connect all resource links since it has performance implications

[the remark this answers is not in the document]
Comment (benjaminapetersen) [reply frontend/public/components/utils/resource-link.jsx]: Right, sorry, spoke too soon.  My local changes are:

```JavaScript 
const NodeLink_ = (props) => {
  const {name, flags: { CAN_LIST_NODE: listNodeFlag }} = props;
  if (!name) {
    return <React.Fragement>-</React.Fragement>;
  }
  return listNodeFlag 
    ? <ResourceLink kind="Node" name={name} title={name} />
    : <React.Fragement>{name}</React.Fragement>;
}
```
So its simplified, not eliminated.

**Comment by benjaminapetersen** [comment]
Events
screen shot 2018-07-20 at 11 47 56 am
Pods
screen shot 2018-07-20 at 11 48 22 am
Containers
screen shot 2018-07-20 at 11 48 38 am

PR is not yet squashed in case we decide not to go with the abbr changes.

**Comment by spadgett** [comment]
/hold
for feedback from @tlwu2013 since this changes some of the resource icons

**Comment by spadgett** [comment]
ping @tlwu2013

**Comment by benjaminapetersen** [comment]
Ping @tlwu2013 

Just curious if you landed on an icon opinion :)

**Comment by tlwu2013** [comment]
Summarize the changes (a bit hard to follow the changes since the latest screenshots are not posted on PR's description):

**[Before this PR]:**
_Pod list:_
pod-list__before

_Event list:_
event__before

**[Changes in this PR]:** change Namespace badge to `NS` and add Node badge as `N`
_Pod list:_
pod-list__pr

_Event list:_
events__pr

>@tlwu2013 We were guessing that we don't show the node resource icon because it's so similar to namespace icon and they tend to be near each other. Is that the reason?

The badge for "Nodes" has already been using acronym `N` (see Node list/details view). The reason that "Pod list" view doesn't show Node's resource badge is because long before the list view only shows the resource badge of the primary object (i.e. the 1st column). This is established before I joined the team, but I guess the reason is to drive the focus on the primary object. (cc @robszumski )

So for that reason, I don't feel there is a strong need to display the badge of Node on Pod list view.
But if for the sake of consistency, I am not against "adding Node badge with `N`, and change the acronym of Namespace to `NS`".

**Comment by spadgett** [comment]
> [Changes in this PR]: change Namespace badge to NS and add Node badge as N

OK, are we good with the change?

For reference, here are the kubectl abbreviations:

https://kubernetes.io/docs/reference/kubectl/overview/#resource-types

kubectl uses `ns` for namespace and `no` for node. Another option is to leave namespace `N` and make node `NO`. I'm OK either way.

**Comment by benjaminapetersen** [comment]
I like `ns` and `no` fine, and consistency is good, but thats just my two cents.

**Comment by robszumski** [comment]
`NS` for namespaces absolutely.

I side towards `N` for nodes, mostly because its already been that way in Tectonic, but I see the point about getting better consistency with `kubectl`.

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
jenkins rebuild
